app/api/v1/endpoints/market_analysis.py

```
@@ -0,0 +1,1025 @@
+    try:
+        # Initialize market data feeds if needed
+        if not hasattr(market_data_feeds, 'redis') or market_data_feeds.redis is None:
+            await market_data_feeds.async_init()
+        
+        result = await market_data_feeds.get_trending_coins(limit=limit)
+        
+        if not result.get("success"):
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=result.get("error", "Failed to get trending coins")
+            )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)